pubsub/rabbitpubsub/rabbit.go

```diff
@@ -564,15 +589,34 @@ func (s *subscription) establishChannel(ctx context.Context) error {
 		if err != nil {
 			return err
 		}
+		// Apply subscription options to channel.
+		err = applyOptionsToChannel(s.opts, ch)
+		if err != nil {
+			return err
+		}
 		// Subscribe to messages from the queue.
 		s.delc, err = ch.Consume(s.queue, s.consumer)
 		return err
 	})
 	if err != nil {
 		return err
 	}
+
 	s.ch = ch
 	s.closec = ch.NotifyClose(make(chan *amqp.Error, 1)) // closec will get at most one element
+
+	return nil
+}
+
+func applyOptionsToChannel(opts *SubscriptionOptions, ch amqpChannel) error {
+	if opts == nil {
```

Comment:
After following my suggestion above, you can remove these 3 lines, `opts` will never be `nil`.

Reply:
done on c47bd5488d052b276f080f8c4130edcbc2aa361d